assets/pages/regular-contributions/helpers/ajax.js

```diff
@@ -94,83 +96,123 @@ function requestData(
     supportAbTests,
   };
 
-  if (state.user.stateField) {
-    regularContribFields.state = state.user.stateField;
+  if (user.stateField) {
+    regularContribFields.state = user.stateField;
   }
 
   return {
     method: 'POST',
-    headers: { 'Content-Type': 'application/json', 'Csrf-Token': state.csrf.token || '' },
+    headers: { 'Content-Type': 'application/json', 'Csrf-Token': csrf.token || '' },
     credentials: 'same-origin',
     body: JSON.stringify(regularContribFields),
   };
 }
 
-function statusPoll(dispatch: Function, getState: Function) {
-  const state = getState();
+let trackingURI = null;
+let pollCount = 0;
```

Comment:
As discussed offline, I realise this was the most practical solution for this PR, but can we look at it again in the future? I'm a little bit concerned about fragmentation of our state/global variables creeping in 😯

Reply:
I strongly agree with your worries, unfortunately, I didn't find a way to avoid this pseudo internal state inside this function. Maybe it is a consequence of trying to have a component without a state (that probably should require one) like StripePopUp Button, interacting with the rest of the app.